src/lib/forward.ts

```diff
@@ -250,6 +254,8 @@ export class EventForwarder {
         sentFragments: number,
         totalFragments: number
       ) => {
+        this.nodeMessageHandler.firmwareUpdateQueued[changedNode.nodeId] =
```

Comment:
It is weird that the event forwarder knows about internal state of the nodeMessageHandler. I would expect nodeMessageHandler subscribe to events themselves.

Reply:
ah that does simplify things...